modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/price/CommerceProductPriceCalculationV2Impl.java

```diff
@@ -395,30 +395,19 @@ public CommerceMoney getUnitPrice(
 			cpInstanceId, quantity, commerceContext);
 	}
 
-	public void unsetCommercePriceListDiscovery(
-		CommercePriceListDiscovery commercePriceListDiscovery,
-		Map<String, Object> properties) {
-
-		String commercePriceListDiscoveryKey = GetterUtil.getString(
-			properties.get("commerce.price.list.discovery.key"));
-
-		_commercePriceListDiscoveryMap.remove(commercePriceListDiscoveryKey);
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
+			bundleContext, CommercePriceListDiscovery.class, null,
+			(serviceReference, emitter) -> emitter.emit(
```

Comment:
@W2jx You should be able to use `com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory#openSingleValueMap(org.osgi.framework.BundleContext, java.lang.Class<S>, java.lang.String)` directly in this pull, if you want to keep the usage of component property and the services are provided by Liferay (provider type).